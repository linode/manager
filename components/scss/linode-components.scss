@import 'mixins';
@import 'variables';
@import 'bootstrap';
@import 'utilities';

@import 'components/index';

html {
  font-size: 62.5%; // resets root font size to 10px
}

// default font sizes, defined in rem with a pixel fallback, computed whole pixel values
.font-normal {
  @include font-size(1.5);
}
.font-small {
  @include font-size(1.3);
}
.font-xs {
  @include font-size(1.2);
}
.font-medium {
  @include font-size(1.6);
}
.font-large {
  @include font-size(1.7);
}
.font-extra-large {
  @include font-size(1.8);
}

body {
  @extend .font-normal;
  background: $background-color;
  color: $black;
  font-weight: normal;
}

h1, h2, h3, h4, h5, h6 {
  margin: 0;
  font-weight: normal;
  line-height: normal;
  color: $black;
}

h1 {
  @include font-size(2.8);
  margin-bottom: 15px;
}

h2 {
  @include font-size(2.2);
}

h3 {
  @extend .font-large;
}

h4 {
  @extend .font-medium;
}

h5, h6 {
  @extend .font-normal;
}

a, a.force-link:not([href]):not([tabindex]) {
    color: $blue;
    text-decoration: none;
    cursor: pointer;

    &:hover, &:focus, &:active {
        text-decoration: underline;
    }
}

a > i.fa {
  color: $blue;
}

label {
  margin-bottom: 0;
}

p {
  margin-bottom: $main-padding-tb/2;
}

hr {
    width: 100%;
    margin: 0;
    background: $gray;
}

small {
  @extend .font-small;
}

strong {
  font-weight: 500; // toned down from 'bolder'
}

section {
    margin-bottom: 15px;
}

.text-muted {
    // Bootstrap sets !important here by default;
    color: $lightest-black !important;
}

.full-height {
    height: 100%;
}

.navicon {
    @extend .font-large;
    padding-left: 15px;
    position: relative;
    top: 2px;

    &:hover,
    &:active,
    &:focus {
        color: $blue;
    }
}

footer {
    @extend .font-small;

    padding: 15px 0;

    span, a {
        color: $lightest-black;
        padding: 0 10px;
    }

    a {
        text-decoration: underline;
    }
}

.Error {
    text-align: center;

    h1 {
        font-size: 8rem; /* Special font-size case */
        color: $dark-gray;
    }

    &-body {
        margin-top: 10px;
    }
}

.sub-header {
    color: $light-black;
    padding-top: 15px;
    padding-bottom: 5px;
}

header .sub-header {
    padding-bottom: 0px;
}

.list-unstyled {
    margin-bottom: 0;
}

.row-justify {
    @extend .row;
    display: flex;
    flex-wrap: wrap;

    margin-left: -$grid-gutter-width-base / 4;
    margin-right: -$grid-gutter-width-base / 4;

    & > [class^="col-"] {
        padding-left: $grid-gutter-width-base / 4;
        padding-right: $grid-gutter-width-base / 4;
    }
}

a:hover, a:focus, a:active {
    color: $blue;
}

.alert {
    @extend .font-xs;

    display: inline-block;
    margin: 5px 0 0 0;
    padding: 15px;
    line-height: initial;
    min-height: initial;
}
