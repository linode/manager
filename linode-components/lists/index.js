export { default as List } from './List';
export { default as ScrollingList } from './ScrollingList';
