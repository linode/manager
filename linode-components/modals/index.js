export { default as ModalShell } from './ModalShell';
export { default as ConfirmModalBody } from './ConfirmModalBody';
export { default as DeleteModalBody } from './DeleteModalBody';
