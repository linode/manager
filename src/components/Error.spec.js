import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';

import { Error } from 'linode-components';


describe('components/Error', () => {
  const sandbox = sinon.sandbox.create();

  afterEach(() => {
    sandbox.restore();
  });

  it('renders a 404 component', () => {
    const error = shallow(<Error status={404} />);

    expect(error.find('h1').text()).toBe('404');
    expect(error.find('h2').text()).toBe('Whoops!');
    expect(error.find('.Error-body').text())
      .toBe('The page you\'re trying to reach does not exist.');
  });

  it('renders a 400 component', () => {
    const error = shallow(<Error status={400} />);

    expect(error.find('h1').text()).toBe('400');
    expect(error.find('h2').text()).toBe('Doh!');
    expect(error.find('.Error-body').text()).toBe('You are not authorized to access this page.');
  });

  it('renders a 500 component', () => {
    const error = shallow(<Error status={500} href="foobar" />);

    expect(error.find('h1').text()).toBe('500');
    expect(error.find('h2').text()).toBe('Uh-oh!');
    expect(error.find({ href: 'foobar' }).text()).toBe('contact support');
  });
});
