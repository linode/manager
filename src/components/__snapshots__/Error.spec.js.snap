// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Error should render without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    href="mailto:support@linode.com"
    status={404}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          404
</h1>,
        <h2>
          Whoops!
</h2>,
        <div
          className="Error-body"
>
          The page you're trying to reach does not exist.
</div>,
      ],
      "className": "Error",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 404,
        },
        "ref": null,
        "rendered": 404,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Whoops!",
        },
        "ref": null,
        "rendered": "Whoops!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The page you're trying to reach does not exist.",
          "className": "Error-body",
        },
        "ref": null,
        "rendered": "The page you're trying to reach does not exist.",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            404
</h1>,
          <h2>
            Whoops!
</h2>,
          <div
            className="Error-body"
>
            The page you're trying to reach does not exist.
</div>,
        ],
        "className": "Error",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 404,
          },
          "ref": null,
          "rendered": 404,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Whoops!",
          },
          "ref": null,
          "rendered": "Whoops!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The page you're trying to reach does not exist.",
            "className": "Error-body",
          },
          "ref": null,
          "rendered": "The page you're trying to reach does not exist.",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
